markdown/handler.php

```diff
@@ -15,12 +15,23 @@
 $legalExtensions = array('md', 'markdown');
 
 $file = realpath($_SERVER['PATH_TRANSLATED']);
-if($file
-	&& in_array(strtolower(substr($file,strrpos($file,'.')+1)), $legalExtensions)
-	&& substr($file,0,strlen($_SERVER['DOCUMENT_ROOT'])) == $_SERVER['DOCUMENT_ROOT']) {
-	echo Markdown(file_get_contents($file));
-} else {
-	echo "<p>Bad filename given</p>";
+
+if (strtoupper(substr(PHP_OS, 0, 3)) === 'WIN') {
```

Comment:
It would be better to reduce the amount of duplicated code:

Eg use this and then refer to $documentRoot in the comparison code
```
if (strtoupper(substr(PHP_OS, 0, 3)) === 'WIN') {
  $documentRoot = str_replace('/', '\\', $_SERVER['DOCUMENT_ROOT']);
} else {
  $documentRoot = $_SERVER['DOCUMENT_ROOT'];
}
```

Or rewrite file:

```
$file = str_replace(DIRECTORY_SEPARATOR, '/', realpath($_SERVER['PATH_TRANSLATED']));
```

Reply:
You have right!!

I will refactorize it.

Reply:
Done!